### DISCUSSION TITLE
Is this a bug? tar_cue_age doesn't seem to work for file targets

### DISCUSSION THREAD

**Discussion by kendonB** [ropensci/tarchetypes#71] [Help]
``` r
targets::tar_dir({ # tar_dir() runs code from a temporary directory.
  targets::tar_script({
    library(tarchetypes)
    list(
      tar_file(
        fn,
        readr::readr_example("mtcars.csv"),
        cue = tarchetypes::tar_cue_age(
          name = fn,
          age = as.difftime(10, units = "secs")
        )
      )
    )
  })
  targets::tar_make()
  Sys.sleep(0.6)
  targets::tar_make()
})
#> • start target fn
#> • built target fn
#> • end pipeline
#> • start target fn
#> • built target fn
#> • end pipeline
```

<sup>Created on 2021-09-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Two other things were a bit strange to me - I couldn't find documentation on `tar_file` beyond the very large help file containing that and many other tarchetypes. I would have thought `tar_file` would have its own.

Then the `name` argument is required for `tar_cue_age` despite it going in as an argument of `tar_file`, requiring the user to put the name in twice. What function does the `name` argument serve in `tar_cue_age`?

**Comment by wlandau** [comment]
Looks like I forgot to update the documentation on metadata timestamps at some point. Fixed in https://github.com/ropensci/targets/commit/f216c466a8b2914d642b64f7469b4e40e3fc2e19 and https://github.com/ropensci/tarchetypes/commit/2513046dee577386f0d0dc0309ea127641ef897.  For local file targets, the time stamp is the modification time of the file. So for local input files that do not change, the age does not reset when you rerun the target.

I find it hard to think of a situation where you would need `tar_cue_age()` for local input files. Rather than enforcing a time delay, would it be better to rely on existing cues to rerun the target  if the hash changes?

`tar_file()` is a thin target factory around `tar_target(format = "file")`, and other storage-format-themed target factories are similar. That's how I thought about it.

`tar_cue_age()` runs independently of the target you assign it to. Cue objects do not have names in and of themselves, and they are resolved statically before the target is defined. That means a target name is required if the cue depends on the target's timestamp. `tar_age()` abstracts this so you do not need to write the name twice.

**Comment by kendonB** [comment]
The use case for me is that the hashing is costly when there are a lot of files or large files and I'm developing a project. The idea was that I'll get the system to check for file updates once a week/day or so, rather than every time tar_make runs.

Using modification time should usually be fine for this use case, but it is possible for a file hash to remain the same with the modification time changing and vice versa.

Is there a cue that would work like this, checking files once a day, for example?

**Reply by wlandau** [reply]
What about `cue = tar_cue(file = tar_older(time = Sys.time() - as.difftime(1, units = "days"), names = target_name))`?

**Reply by wlandau** [reply]
Hmm..., maybe not that one...

**Reply by wlandau** [reply]
You could use `tar_cue_age()` on the time stamp of a non-input-file target immediately downstream.

```{r}
list(
  tar_target(
    big_file,
    "big_file.csv",
    format = "file",
    cue = tar_cue_age(age = as.difftime(1, units = "days"), name = downstream_target)
  ),
  tar_target(downstream_target, big_file)
)
```
